NC_TweenUI.js

```diff
@@ -89,39 +93,85 @@ function NC_Tween() {
 
 	this.getManipulationSelection = function( ){
 
+		//the types of nodes that we want to be manipulating ( all deformers + pegs )
+		nodeTypes_toManipulate = this.deformationTypes
+		Array.prototype.push.apply(nodeTypes_toManipulate, ["PEG"] )
+
 		//create and empty array to put the nodes we want to manipulate into
 		manipulationSelection = new Array(0)
 
 		var initialSelection = selection.selectedNodes();
+
 		if( initialSelection.length == 0 ){
 			this.NC_Log("NC_Tween.getManipulationSelection : no nodes selected ")
 			return
 		}
 		else{
-			//the types of nodes that we want to be manipulating
-			nodeTypes_toManipulate = ["PEG", 
-								// all the other tweenable nodes we might want to deal with
-								"CurveModule", "OffsetModule", "DeformationCompositeModule", "KinematicOutputModule", "TransformationSwitch", "PointConstraint2","REMOVE_TRANSPARENCY","BendyBoneModule","ArticulationModule","BezierMesh","GameBoneModule","GLUE","REFRACT","BoneModule","Turbulence","DeformationRootModule","DeformationScaleModule","DeformationSwitchModule","DeformationUniformScaleModule","DeformationWaveModule","FoldModule","AutoFoldModule","AutoMuscleModule"]
-
+			
+			// check if the selection is all drawings, if so then they user (probably) selected in the camera view so we should make sure the deformer/peg imediately above the drawing are being manipulated
+			var selectionIsAllDrawings = true
 			for( i in initialSelection){
+				selNode = initialSelection[i]
+				if(node.type(selNode) != "READ"){
+					selectionIsAllDrawings = false
+				}
+			}
 
-				selNode 			= initialSelection[i]
-				selNode_type 		= node.type(selNode)
+			if ( selectionIsAllDrawings ){
+				
+				for(d in initialSelection){
+					selDrawing 	= initialSelection[d]
+					//get the drawings parent
+					selParent 		= node.srcNode(selDrawing,0)
+					selParent_type = node.type(selParent)
+
+					// if the parent is a peg then add it to the selection
```

Comment:
This works well if there's no deformer on the drawing, if there's a deformer there, then it doesn't keep going up the hierarchy to find the parent peg.

Is it possible to check and see if the parent is a group, and if so it will add the deformers, and also go up the hierarchy one more and grab the parent peg?

Reply:
I updated it to do exactly that.